### PR TITLE
MRG: Add "see also-sss"

### PR DESCRIPTION
**Pull request by mainakjas**
closes https://github.com/mne-tools/mne-python/issues/2289

It's WIP but I made a PR so that people can jump in and make suggestions for related methods :)

### CODE REVIEW

File: mne/evoked.py [hunk 1]
```diff
@@ -819,6 +819,10 @@ class EvokedArray(Evoked):
     verbose : bool, str, int, or None
         If not None, override default verbose level (see mne.verbose).
         Defaults to raw.verbose.
+
+    See also
```
Comment (teonlamont) [inline mne/evoked.py]: also --> Also

File: mne/cov.py [hunk 2]
```diff
@@ -525,6 +529,10 @@ def compute_covariance(epochs, keep_sample_mean=True, tmin=None, tmax=None,
         returned (sorted by log-likelihood, from high to low, i.e. from best
         to worst).
 
+    See Also
+    --------
+    compute_raw_data_covariance : Estimate noise cov from raw data
```
Comment (teonlamont) [inline mne/cov.py]: you say cov here, but just above, it is written covariance. should keep the consistency

File: mne/epochs.py [hunk 3]
```diff
@@ -879,6 +879,10 @@ def drop_log_stats(self, ignore=['IGNORED']):
         -------
         perc : float
             Total percentage of epochs dropped.
+
+        See Also
+        --------
+        plot_drop_log
```
Comment (teonlamont) [inline mne/epochs.py]: for some of the functions, you have a brief blurb, but for some of the others, you don't.

File: mne/minimum_norm/inverse.py [hunk 4]
```diff
@@ -103,6 +103,10 @@ def read_inverse_operator(fname, verbose=None):
     -------
     inv : instance of InverseOperator
         The inverse operator.
+
+    See Also
+    --------
+    write_inverse_operator
```
Comment (teonlamont) [inline mne/minimum_norm/inverse.py]: like here

File: mne/minimum_norm/inverse.py [hunk 5]
```diff
@@ -323,6 +327,10 @@ def write_inverse_operator(fname, inv, verbose=None):
         The inverse operator.
     verbose : bool, str, int, or None
         If not None, override default verbose level (see mne.verbose).
+
+    See Also
+    --------
+    read_inverse_operator
```
Comment (teonlamont) [inline mne/minimum_norm/inverse.py]: and here

File: mne/epochs.py [hunk 6]
```diff
@@ -1720,6 +1724,10 @@ class EpochsArray(_BaseEpochs):
     verbose : bool, str, int, or None
         If not None, override default verbose level (see mne.verbose).
         Defaults to raw.verbose.
+
+    See Also
+    --------
+    io.RawArray, mne.EvokedArray
```
Comment (teonlamont) [inline mne/epochs.py]: maybe they are needed here as well

File: mne/epochs.py (hunk 3 above)
Comment (mainakjas) [inline mne/epochs.py]: that's intentional. See this: https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt

Comment (mainakjas) [inline mne/epochs.py]: I guess I didn't put the brief blurb where I thought it was obvious or when I was too lazy :P

File: mne/time_frequency/tfr.py [hunk 7]
```diff
@@ -266,6 +270,10 @@ def cwt_morlet(X, sfreq, freqs, use_fft=True, n_cycles=7.0, zero_mean=False):
     -------
     tfr : 3D array
         Time Frequency Decompositions (n_signals x n_frequencies x n_times)
+
+    See Also
+    --------
+    morlet : compute Wavelets for a frequency range
     """
     mode = 'same'
```
Comment (dengemann) [inline mne/time_frequency/tfr.py]: stockwell?

File: mne/cov.py (hunk 2 above)
Comment (agramfort) [inline mne/cov.py]: +1 for covariance

File: mne/epochs.py (hunk 6 above)
Comment (agramfort) [inline mne/epochs.py]: I am not sure the blurb is needed.

File: mne/minimum_norm/inverse.py [hunk 8]
```diff
@@ -754,6 +762,11 @@ def apply_inverse(evoked, inverse_operator, lambda2=1. / 9.,
     -------
     stc : SourceEstimate | VolSourceEstimate
         The source estimates
+
+    See Also
+    --------
+    apply_inverse_raw : Apply inverse operator to raw object
+    apply_inverse_epochs : Apply inverse operator to epochs object
```
Comment (agramfort) [inline mne/minimum_norm/inverse.py]: it's mostly for cross refs in the doc. Just check it renders fine.

**Comment by agramfort** [comment]
thanks @mainakjas for tackling this. This is very useful. Let us know when you're done.

**Comment by Eric89GXL** [comment]
@mainakjas how close are we on this? It would be good to get this merged sooner rather than later since it's going to touch a lot of functions, and could therefore cause rebasing problems with other PRs.

**Comment by mainakjas** [comment]
I could do with one round of reviews.

**Comment by Eric89GXL** [comment]
Have you checked to make sure these all look good when built (e.g., comma-separated functions look good / link properly)?

File: mne/beamformer/_rap_music.py [hunk 9]
```diff
@@ -224,6 +224,10 @@ def rap_music(evoked, forward, noise_cov, n_dipoles=5, return_residual=False,
         The residual a.k.a. data not explained by the dipoles.
         Only returned if return_residual is True.
 
+    See Also
+    --------
+    mne.Dipole.fit_dipole
```
Comment (Eric89GXL) [inline mne/beamformer/_rap_music.py]: can you get away with just `fit_dipole` here? If you can't, it should be `mne.dipole.fit_dipole` or probably just `mne.fit_dipole` (I think we import it there)

File: mne/bem.py [hunk 10]
```diff
@@ -986,6 +986,10 @@ def read_bem_surfaces(fname, patch_stats=False, s_id=None, verbose=None):
     surf: list | dict
         A list of dictionaries that each contain a surface. If s_id
         is not None, only the requested surface will be returned.
+
+    See Also
+    --------
+    write_bem_surfaces
```
Comment (Eric89GXL) [inline mne/bem.py]: read_bem_solution, write_bem_solution, make_bem_model

File: mne/bem.py [hunk 11]
```diff
@@ -1115,6 +1119,10 @@ def read_bem_solution(fname, verbose=None):
     -------
     bem : dict
         The BEM solution.
+
+    See Also
+    --------
+    write_bem_solution
```
Comment (Eric89GXL) [inline mne/bem.py]: read_bem_surfaces, write_bem_surfaces, make_bem_solution

File: mne/channels/layout.py [hunk 12]
```diff
@@ -145,6 +149,10 @@ def read_layout(kind, path=None, scale=True):
     -------
     layout : instance of Layout
         The layout.
+
+    See Also
+    --------
+    save
```
Comment (Eric89GXL) [inline mne/channels/layout.py]: This probably needs to be `mne.Layout.save` or whatever the class name is to be rendered properly (please check all refs render properly to the correct links)

File: mne/cov.py [hunk 13]
```diff
@@ -236,6 +236,10 @@ def read_cov(fname, verbose=None):
     -------
     cov : Covariance
         The noise covariance matrix.
+
+    See Also
+    --------
+    write_cov
```
Comment (Eric89GXL) [inline mne/cov.py]: compute_covariance, compute_raw_data_covariance

File: mne/epochs.py (hunk 6 above)
Comment (Eric89GXL) [inline mne/epochs.py]: Either `mne.io.RawArray` and `mne.EvokedArray` or `io.RawArray` and `EvokedArray`. Hopefully the latter will work fine

File: mne/event.py [hunk 14]
```diff
@@ -215,6 +215,10 @@ def read_events(filename, include=None, exclude=None, mask=0):
     events: array, shape (n_events, 3)
         The list of events
 
+    See Also
+    --------
+    write_events
```
Comment (Eric89GXL) [inline mne/event.py]: find_events

File: mne/evoked.py [hunk 15]
```diff
@@ -819,6 +819,10 @@ class EvokedArray(Evoked):
     verbose : bool, str, int, or None
         If not None, override default verbose level (see mne.verbose).
         Defaults to raw.verbose.
+
+    See Also
+    --------
+    epochs.EpochsArray, io.RawArray
```
Comment (Eric89GXL) [inline mne/evoked.py]: pretty sure `EpochsArray` is at the base level

File: mne/filter.py [hunk 16]
```diff
@@ -626,6 +626,10 @@ def band_pass_filter(x, Fs, Fp1, Fp2, filter_length='10s',
     xf : array
         x filtered.
 
+    See Also
+    --------
+    low_pass_filter, high_pass_filter
```
Comment (Eric89GXL) [inline mne/filter.py]: resample

File: mne/forward/_make_forward.py [hunk 17]
```diff
@@ -352,6 +352,10 @@ def make_forward_solution(info, trans, src, bem, fname=None, meg=True,
     fwd : instance of Forward
         The forward solution.
 
+    See Also
+    --------
+    forward.do_forward_solution
```
Comment (Eric89GXL) [inline mne/forward/_make_forward.py]: shouldn't need to prepend `forward.`

read_forward_solution, write_forward_solution

File: mne/forward/forward.py [hunk 18]
```diff
@@ -423,6 +423,10 @@ def read_forward_solution(fname, force_fixed=False, surf_ori=False,
     -------
     fwd : instance of Forward
         The forward solution.
+
+    See Also
+    --------
+    write_forward_solution
```
Comment (Eric89GXL) [inline mne/forward/forward.py]: make_forward_solution

File: mne/io/pick.py [hunk 19]
```diff
@@ -84,6 +84,10 @@ def pick_channels(ch_names, include, exclude=[]):
         List of channels to exclude (if empty do not exclude any channel).
         Defaults to [].
 
+    See Also
+    --------
+    pick_channels_regexp
```
Comment (Eric89GXL) [inline mne/io/pick.py]: pick_types

File: mne/minimum_norm/inverse.py (hunk 4 above)
Comment (Eric89GXL) [inline mne/minimum_norm/inverse.py]: make_inverse_operator

File: mne/preprocessing/ica.py [hunk 20]
```diff
@@ -887,6 +887,10 @@ def find_bads_ecg(self, inst, ch_name=None, threshold=None,
         scores : np.ndarray of float, shape (``n_components_``)
             The correlation scores.
 
+        See also
+        --------
+        find_bads_eog
```
Comment (Eric89GXL) [inline mne/preprocessing/ica.py]: might need to be `ICA.find_bads_eog`, did you check this one renders okay?

File: mne/source_space.py [hunk 21]
```diff
@@ -511,6 +511,10 @@ def read_source_spaces(fname, patch_stats=False, verbose=None):
     -------
     src : SourceSpaces
         The source spaces.
+
+    See Also
+    --------
+    write_source_spaces
```
Comment (Eric89GXL) [inline mne/source_space.py]: setup_source_space, setup_volume_source_space

File: mne/surface.py [hunk 22]
```diff
@@ -670,6 +674,10 @@ def write_surface(fname, coords, faces, create_stamp=''):
     create_stamp : str
         Comment that is written to the beginning of the file. Can not contain
         line breaks.
+
+    See Also
+    --------
+    write_surface
```
Comment (Eric89GXL) [inline mne/surface.py]: I think you meant read_surface

**Comment by Eric89GXL** [comment]
Looks like a good start, but I'd try to make things more consistent. I'd rather have no descriptions, and omit `mne.` from all names if possible since it's implied.

**Comment by agramfort** [comment]
we should just make sure it renders fine in the doc with cross links.

File: mne/bem.py (hunk 11 above)
Comment (mainakjas) [inline mne/bem.py]: btw, I was trying to figure out with @wronk what's the difference between `make_bem_model` and `make_bem_solution`. Could you throw some light?

File: mne/channels/layout.py (hunk 12 above)
Comment (mainakjas) [inline mne/channels/layout.py]: The `Layout` class doesn't seem to be listed in the API reference page. It should be, right?

Comment (mainakjas) [inline mne/channels/layout.py]: same question for `Montage` and `DigMontage` class

File: mne/filter.py (hunk 16 above)
Comment (mainakjas) [inline mne/filter.py]: doesn't sound very related to me ... so I'll skip this for now. But thanks for the suggestion!

File: mne/bem.py (hunk 11 above)
Comment (Eric89GXL) [inline mne/bem.py]: Read our revamped manual :)

But really, the `model` downsamples and puts together one or three surfaces with conductances in a FIF file after doing some sanity checks. The `solution` actually computes the information based on those surfaces and conductances that we need for forward computation.

File: mne/filter.py (hunk 16 above)
Comment (Eric89GXL) [inline mne/filter.py]: In signal processing terms resampling can be thought of as low-pass filtering followed by a decimation step (or a zero-insertion step followed by a low-pass)

File: mne/channels/layout.py (hunk 12 above)
Comment (agramfort) [inline mne/channels/layout.py]: oups that's a bug... It should be there.

**Comment by agramfort** [comment]
let's merge after adressing @Eric89GXL 's comments. It's already a big improvement.

File: mne/filter.py (hunk 16 above)
Comment (mainakjas) [inline mne/filter.py]: ok got it :) I'll add it.

File: mne/preprocessing/ica.py (hunk 20 above)
Comment (mainakjas) [inline mne/preprocessing/ica.py]: yeah, this renders fine :)

**Comment by mainakjas** [comment]
I think I'm good here

**Comment by agramfort** [comment]
merged by rebase. Thanks @mainakjas !

**Pull request closed without merging**